File: src/main/java/com/github/chMatvey/dronedelivery/model/DroneModel.java
package com.github.chMatvey.dronedelivery.model;

import lombok.Getter;
import lombok.RequiredArgsConstructor;

@Getter
@RequiredArgsConstructor
public enum DroneModel {
    LIGHTWEIGHT("Lightweight"),
    MIDDLEWEIGHT("Middleweight"),
    CRUISERWEIGHT("Cruiserweight"),
    HEAVYWEIGHT("Heavyweight");

    private final String value;
}

File: src/main/kotlin/com/github/chMatvey/models/DroneModel.kt
package com.github.chMatvey.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
enum class DroneModel(val value: String) {
    @SerialName("Lightweight")
    LIGHTWEIGHT("Lightweight"),

    @SerialName("Middleweight")
    MIDDLEWEIGHT("Middleweight"),

    @SerialName("Cruiserweight")
    CRUISERWEIGHT("Cruiserweight"),

    @SerialName("Heavyweight")
    HEAVYWEIGHT("Heavyweight")
}

fun toDroneModel(value: String): DroneModel {
    return DroneModel.entries.firstOrNull { it.value == value }
        ?: throw IllegalArgumentException("Unknown drone model: $value")
}